modules/styleguide/assets/js/frontend/contexts/settings.js

```diff
@@ -0,0 +1,164 @@
+				[ 'is_debug', elementorCommon.config.isElementorDebug ],
+			] ) ],
+		] );
+
+		setSettings( map );
+	};
+
+	const onCommandEvent = ( event ) => {
+		switch ( event.detail.command ) {
+			case 'document/elements/settings':
+				onSettingsChange( event.detail.args );
+				break;
+			case 'document/repeater/insert':
+				onInsert( event.detail.args );
+				break;
+			case 'document/repeater/remove':
+				onRemove( event.detail.args );
+				break;
+			default:
+				break;
+		}
+	};
+
+	/**
+	 * Triggered when a color or font is changed.
+	 * Has a 100ms debounce.
+	 *
+	 * @param {{container: {model: {attributes: {name: string}}, id: number}, settings: {}}} args
+	 */
+	const onSettingsChange = useDebouncedCallback( ( args ) => {
+		const name = args.container.model.attributes.name;
+
+		const newSettings = new Map( settingsRef.current );
+
+		for ( const [ group, groupSettings ] of newSettings.entries() ) {
+			if ( ! groupSettings.has( name ) ) {
+				continue;
+			}
+
+			if ( Array.isArray( groupSettings.get( name ) ) ) {
+				const index = groupSettings.get( name ).findIndex( ( item ) => item._id === args.container.id );
+
+				if ( -1 === index ) {
+					return;
+				}
+
+				newSettings.get( group ).get( name )[ index ] = { ...groupSettings.get( name )[ index ], ...args.settings };
+			} else {
+				newSettings.get( group ).set( name, args.settings );
+			}
+		}
+
+		setSettings( newSettings );
+	}, 100 );
+
+	/**
+	 * Triggered when a new custom color or font is created.
+	 *
+	 * @param {{name: string, model: string, options: {at: number}}} args
+	 */
+	const onInsert = ( args ) => {
+		const name = args.name;
+
+		const newSettings = new Map( settingsRef.current );
+
+		for ( const [ group, groupSettings ] of newSettings.entries() ) {
+			if ( ! groupSettings.has( name ) ) {
+				continue;
+			}
+
+			const newArray = [ ...groupSettings.get( name ) ];
+
+			const at = ( undefined === args.options?.at )
+				? newArray.length
+				: args.options.at;
+
+			newSettings.get( group ).set( name, [ ...newArray.slice( 0, at ), args.model, ...newArray.slice( at ) ] );
+		}
+
+		setSettings( newSettings );
+	};
+
+	/**
+	 * Triggered when a custom color or font is removed.
+	 *
+	 * @param {{name: string, index: number}} args
+	 */
+	const onRemove = ( args ) => {
+		const name = args.name;
+
+		const newSettings = new Map( settingsRef.current );
+
+		for ( const [ group, groupSettings ] of newSettings.entries() ) {
+			if ( ! groupSettings.has( name ) ) {
+				continue;
+			}
+
+			const newArray = [ ...groupSettings.get( name ) ];
+
+			newSettings.get( group ).set( name, newArray.filter( ( item, index ) => index !== args.index ) );
+		}
+
+		setSettings( newSettings );
+	};
+
+	useEffect( () => {
+		getInitialSettings();
+
+		window.top.addEventListener( 'elementor/commands/run/after', onCommandEvent, { passive: true } );
+
+		return () => {
+			window.top.removeEventListener( 'elementor/commands/run/after', onCommandEvent );
```

Comment:
onCommandEvent has a new reference in each render, so it seems like this code doesn't work.

Reply:
You are right, pushed a fix.

Can you please check again and resolve?